aten/src/ATen/native/cpu/BinaryOpsKernel.cpp

```diff
@@ -128,7 +128,7 @@ void remainder_kernel(TensorIterator& iter) {
       });
     });
   } else {
-    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "remainder_cpu", [&]() {
+    AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.common_dtype(), "remainder_cpu", [&]() {
```

Comment:
Move this dispatch out of the else branch.

Reply:
I think we still need to keep it within the else block. Otherwise, we will dispatch two times for this operation.